target/linux/generic/pending-5.10/800-bcma-get-SoC-device-struct-copy-its-DMA-params-to-th.patch

```diff
@@ -50,20 +50,14 @@ Signed-off-by: Rafał Miłecki <rafal@milecki.pl>
  	if (err)
 --- a/drivers/bcma/main.c
 +++ b/drivers/bcma/main.c
-@@ -236,12 +236,16 @@ EXPORT_SYMBOL(bcma_core_irq);
- 
- void bcma_prepare_core(struct bcma_bus *bus, struct bcma_device *core)
- {
-+	struct device *dev = &core->dev;
-+
- 	core->dev.release = bcma_release_core_dev;
+@@ -241,8 +241,10 @@ void bcma_prepare_core(struct bcma_bus *
  	core->dev.bus = &bcma_bus_type;
  	dev_set_name(&core->dev, "bcma%d:%d", bus->num, core->core_index);
  	core->dev.parent = bus->dev;
 -	if (bus->dev)
 +	if (bus->dev) {
  		bcma_of_fill_device(bus->dev, core);
-+		dma_coerce_mask_and_coherent(dev, bus->dev->coherent_dma_mask);
++		dma_coerce_mask_and_coherent(&core->dev, bus->dev->coherent_dma_mask);
```

Comment:
Are you sure the "&" is correct here? Previously is was assigned to `*dev`, but now it's replacing plain `dev`. I'd have expected just "core->dev" here. But I'm only looking at this with my limited C++ skills...

Reply:
Yes this is correct. 
The line above  does it in a similar way:
```
dev_set_name(&core->dev, "bcma%d:%d", bus->num, core->core_index);
```
`dev` is not a pointer in `core`, but directly embedded in the srtucture.